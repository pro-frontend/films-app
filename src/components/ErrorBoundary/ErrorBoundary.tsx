import React, { Component, ReactNode } from "react";
import DefaultTemplate from "./DefaultTemplate";
import {
  IErrorBoundaryProps,
  IErrorBoundaryState,
} from "./ErrorBoundary.types";

class ErrorBoundary extends Component<
  IErrorBoundaryProps,
  IErrorBoundaryState
> {
  constructor(props: IErrorBoundaryProps) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(): IErrorBoundaryState {
    return { hasError: true };
  }

  render(): ReactNode {
    if (this.state.hasError) {
      return <DefaultTemplate />;
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
