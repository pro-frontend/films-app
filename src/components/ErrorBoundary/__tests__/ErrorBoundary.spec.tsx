import React from "react";
import ErrorBoundary from "../";
let correctWrapper, wrongWrapper: ReactWrapper;

describe("<ErrorBoundary />", () => {
  beforeEach(() => {
    correctWrapper = mount(
      <ErrorBoundary>
        <h2>Correct component</h2>
      </ErrorBoundary>,
    );
    wrongWrapper = mount(
      <ErrorBoundary>
        <h2>{new Error("Component has a mistake")}</h2>
      </ErrorBoundary>,
    );
  });

  it("should mount", () => {
    expect(correctWrapper).toMatchSnapshot();
    expect(wrongWrapper).toMatchSnapshot();
  });
});
