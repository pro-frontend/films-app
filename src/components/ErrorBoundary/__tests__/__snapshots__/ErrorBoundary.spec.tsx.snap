// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> should mount 1`] = `undefined`;

exports[`<ErrorBoundary /> should mount 2`] = `undefined`;
